pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/StringSchema.java

```diff
@@ -37,19 +37,16 @@ public StringSchema(Charset charset) {
         this.charset = charset;
     }
 
-    public byte[] encode(String message) {
-        return message.getBytes(charset);
+    public ByteBuffer encode(String message) {
+        byte[] bytes = message.getBytes(charset);
+        return ByteBuffer.allocateDirect(bytes.length).put(bytes);
     }
 
-    public String decode(byte[] bytes) {
-        return new String(bytes, charset);
+    public String decode(ByteBuffer buf) {
+        return new String(buf.array(), charset);
     }
 
     public SchemaInfo getSchemaInfo() {
-        SchemaInfo schemaInfo = new SchemaInfo();
-        schemaInfo.setName("String");
-        schemaInfo.setType(SchemaType.STRING);
-        schemaInfo.setSchema(new byte[0]);
-        return schemaInfo;
+        return null;
```

Comment:
I think we should retain the schema info. It's marking that a particular topic is using "String" as the type.